### ISSUE TITLE
5.0.300 dotnet restore ignores proxy settings

### ISSUE THREAD

**Issue by jirikopecky** [dotnet/sdk#17991]
Our build server is behind proxy and we use private NuGet feed. This feed have to be accessed directly, without the proxy. Feed DNS name is configured in `no_proxy` environment variable. `dotnet restore` fails with HTTP 504 Gateway Timeout.

Summaray:
- Environment: Docker (using `mcr.microsoft.com/dotnet/sdk:5.0` image)
- Custom NuGet feed configured in NuGet.config (nuget.mycorp.com)
- HTTP(S) proxy configured using `http_proxy` and `https_proxy` env variables
- Using `no_proxy` env variable to configure direct access to custom feed (`no_proxy=nuget.mycorp.com`)
- Restore works with SDK 5.0.203, same setup fails with SDK 5.0.300

**Comment by tempel3** [comment]
Hello,

i have a similar issue.

While trying to setup a proxy WITH Authentication for the docker build using the nuget.config and environmental variables. The settings are ignored or not handled correctly.

For more information regarding the test setup, please ask.

Best regards
Paul

**Comment by protonyx** [comment]
I noticed this same issue with SDK v5.0.300. If I clear my proxy environment variables (`HTTP_PROXY`, `HTTPS_PROXY`, `NO_PROXY`), `dotnet restore` begins working (we also have a private feed on our intranet). It seems that the `NO_PROXY` environment variable is being ignored, even when I specify it in a solution-local nuget.config:

```
  <config>
    <add key="http_proxy" value="http://proxy.mycorp.com" />
    <add key="no_proxy" value=".mycorp.com" />
  </config>
```

**Comment by michal-ciechan** [comment]
Adding from https://github.com/NuGet/Home/issues/6981#issuecomment-856852714

_________

My issue is that we need to use proxy for a single source. and not for others. 

We have HTTP_PROXY set so that everything picks it up correctly. 

We also have NO_PROXY set for various domains/IP ranges that cannot go through the proxy.

Global `NuGet.config`

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <!-- This needs to go through Proxy -->
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />

    <!-- This cannot go through Proxy. Has exclusion in NO_PRXY -->
    <add key="TeamCityInternal" value="http://nuget.internal/" />
  </packageSources>
</configuration>
```

And then we have 

```
HTTP_PROXY=http://somproxy
NO_PROXY=.internal
```

This works perfectly fine when running `nuget restore` using `nuget.exe` versions `v5.9.1` or  `v5.10.0-preview2`

It doesn't work when running `dotnet restore` using .NET SDK `5.0.300` or .NET SDK `6.0.100-preview.4`.

Using `dotnet restore` get a 403 forbidden error. (Proxy uses standard NTLM)

**Comment by michal-ciechan** [comment]
`5.0.301` still broken

**Comment by sfoslund** [comment]
FYI @nkolev92

**Comment by nkolev92** [comment]
Dup in https://github.com/NuGet/Home/issues/10902. 

cc @erdembayar who fixed it. 

fyi @aortiz-msft, we might need to consider patching this.

**Comment by nkolev92** [comment]
For reference, the fix *is available* in the 5.0.400 dailies, https://github.com/dotnet/sdk/pull/18041.

**Comment by erdembayar** [comment]
Here is link for daily build if someone wants to download 5.0.400 daily build
https://github.com/dotnet/installer/blob/main/README.md#installers-and-binaries

**Comment by michal-ciechan** [comment]
this is still broken in VS `16.10` and `16.11 preview2` as those use a broken SDK version ignoring what is installed on the machine or global.json

**Comment by aortiz-msft** [comment]
@michal-ciechan: Could you please try 16.11 preview 3 and letting us know the outcome?

**Comment by erdembayar** [comment]
> @michal-ciechan: Could you please try 16.11 preview 3 and letting us know the outcome?

@aortiz-msft 
He already confirmed problem is fixed with 16.11 Preview 2, also I manually verified it yesterday.
https://github.com/NuGet/Home/issues/10932#issuecomment-875514246

**Comment by jirikopecky** [comment]
Hello, I can confirm that SDK 5.0.400 fixed the issue. Thanks!
